clients/roscpp/include/ros/steady_timer.h

```diff
@@ -67,9 +67,11 @@ class ROSCPP_DECL SteadyTimer
 
   /**
    * \brief Set the period of this timer
+   * \param reset Whether to reset the timer. If true, timer ignores elapsed time and next cb occurs at now()+period
    */
   void setPeriod(const WallDuration& period, bool reset=true);
 
+  bool hasStarted() const { return impl_->hasStarted(); }
```

Comment:
should probably be
```suggestion
  bool hasStarted() const { return impl_ && impl_->hasStarted(); }
```

Reply:
I currently do not see how a `SteadyTimer`, `WallTimer` or `Timer` instance can exist with an invalid `impl_` pointer, because it is always initialized in the constructor and never reset until destruction. But you and @kunaltyagi are right that either all or none of the methods must have a nullptr check.

I will update this PR after #1541 gets merged.

Reply:
#1541 was merged...

Reply:
Updated.

Reply:
You are right. I did not notice the definition in the header.

The three headers are completely identical with this patch, apart from type names, includes and copyright headers. In combination with #1557 the sources will be identical, too.